src/test/shared/clients/defaultLambdaClient.test.ts

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DefaultLambdaClient, LambdaClient } from '../../../shared/clients/lambdaClient'
+import { stub, createStubInstance, SinonStubbedInstance } from 'sinon'
+import { Lambda, Request } from 'aws-sdk'
+import * as assert from 'assert'
+import { FunctionUrlConfigList, FunctionUrlConfig, ListFunctionUrlConfigsResponse } from 'aws-sdk/clients/lambda'
+
+describe('Test the default lambda client', async () => {
+    let client: LambdaClient
+    const regionCode: string = 'my-region-code'
+    const lambdaFuncName = 'lambda-func-name'
+    let lambdaApiMock: SinonStubbedInstance<Lambda>
+
+    beforeEach(async () => {
+        lambdaApiMock = stub(new Lambda())
+        client = new DefaultLambdaClient(regionCode)
+    })
+
+    describe('Test method getFunctionUrlConfigs()', async () => {
+        /**
+         * Stubs the listFunctionUrlConfigs() method with the given
+         * value.
+         * @param functionUrlConfigs
+         */
+        function stubListFunctionUrlConfigs(functionUrlConfigs: FunctionUrlConfigList): void {
+            const functionUrlConfigsList: ListFunctionUrlConfigsResponse = {
+                FunctionUrlConfigs: functionUrlConfigs,
+            }
+            const requestStub = createStubInstance(Request)
+            requestStub.promise.resolves(functionUrlConfigsList)
+            lambdaApiMock.listFunctionUrlConfigs.returns(requestStub as any)
+        }
+
+        const testCases: FunctionUrlConfigList[] = [
+            [],
+            [buildFunctionUrlConfig({ FunctionUrl: 'url1' })],
+            [buildFunctionUrlConfig({ FunctionUrl: 'url1' }), buildFunctionUrlConfig({ FunctionUrl: 'url2' })],
+        ]
+
+        testCases.forEach(functionUrlConfigs => {
+            it('gets the expected url config list', async () => {
+                stubListFunctionUrlConfigs(functionUrlConfigs)
+
+                // Call to test
+                const result = await client.getFunctionUrlConfigs(lambdaFuncName, <Lambda>(<unknown>lambdaApiMock))
```

Comment:
similar to Jaden's comment above, this isn't testing enough to justify the cost of the test. It's good to exercise codepaths however we can, but in this case we should wait until we have a general strategy for testing SDKs.

The code in `getFunctionUrlConfigs` is quite small. It's not sustainable to maintain so much test code without a proportionate gain. If the code in `getFunctionUrlConfigs` were non-trivial, typically I would suggest changing its interface to take SDK result and pass that in from test, so that we don't need all this mocking in the test.

tl;dr: let's drop this test file for now. We will eventually take time to find a way to maintainably test SDK client code.

Reply:
Makes sense to me, I've dropped the file.